addons/hr_holidays/tests/test_allocations.py

```diff
@@ -372,3 +371,58 @@ def test_employee_holidays_archived_display(self):
         employee._compute_allocation_remaining_display()
 
         self.assertEqual(employee.allocation_display, '0')
+
+    def test_refuse_validated_allocation_created_after_leave(self):
+        """
+        Test that an allocation created after a leave can be refused after being validated only if the leave hasn't started yet
+
+        If the related leave has not started, the allocation can be refused
+        If the related leave has already started, refusing the allocation should raise a UserError
+        """
+        leave_employee = self.env['hr.employee'].create({
+            'name': 'Test Employee',
+            'user_id': self.env.uid,
+        })
+
+        allocation = self.env['hr.leave.allocation'].create({
+            'name': 'Initial Allocation',
+            'holiday_status_id': self.leave_type_paid.id,
+            'number_of_days': 20,
+            'employee_id': leave_employee.id,
+            'date_from': date(2025, 1, 1),
+        })
+        allocation.action_validate()
+
+        leave_request = self.env['hr.leave'].create({
+            'name': 'Leave Request',
+            'holiday_status_id': self.leave_type_paid.id,
+            'request_date_from': date.today() + timedelta(days=5),
+            'request_date_to': date.today() + timedelta(days=10),
+            'employee_id': leave_employee.id,
+        })
+
+        post_leave_allocation = self.env['hr.leave.allocation'].create({
+            'name': 'Post Leave Allocation',
+            'holiday_status_id': self.leave_type_paid.id,
+            'number_of_days': 0.5,
+            'employee_id': leave_employee.id,
+            'date_from': date.today(),
+            'date_to': date.today() + timedelta(days=15),
+        })
+        post_leave_allocation.action_validate()
+        self.assertEqual(post_leave_allocation.state, 'validate')
+
+        post_leave_allocation.action_refuse()
+        self.assertEqual(post_leave_allocation.state, 'refuse')
+
+        leave_request.write({
+            'request_date_from': date.today(),
+            'request_date_to': date.today() + timedelta(days=4),
+        })
+
+        post_leave_allocation.action_validate()
+        self.assertEqual(post_leave_allocation.state, 'validate')
+
+        with self.assertRaises(UserError):
+            post_leave_allocation.action_refuse()
+            self.assertEqual(post_leave_allocation.state, 'validate')
```

Comment:
not sure of what we're supposed to test here? 🤔 Maybe also ensure that both allocations can't be refused at the same time